### PR TITLE
Simpler "plumbing"

### PR DESCRIPTION
**Pull request by AlanGriffiths**

### CODE REVIEW

File: src/server/input/default_input_device_hub.cpp [hunk 1]
```diff
@@ -492,6 +496,11 @@ auto mi::DefaultInputDeviceHub::add_device(std::shared_ptr<InputDevice> const& d
         seat->add_device(*handle);
         dev->start(seat, input_dispatchable);
 
+        if (auto const observer = std::dynamic_pointer_cast<LedObserver>(device))
+        {
+            led_observer_registrar->register_interest(observer, handle->id());
+        }
```
Comment (mattkae) [inline src/server/input/default_input_device_hub.cpp:502]: We will need to `unregister_interest` in `remove_device`, right?

**Review by mattkae — CHANGES_REQUESTED** [review]
Makes sense with a lot less plumbing required! I think we just need to unregister_interest, which I will do. Testing this now

**Review by mattkae — APPROVED** [review]
Looks good to me!

File: src/server/input/default_input_device_hub.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply src/server/input/default_input_device_hub.cpp:502]: > We will need to `unregister_interest` in `remove_device`, right?

Oh, I was misled by "register" taking a weak_ptr. That made me think that EOL was an implicit deregister

**Pull request merged**